pkg/podimpersonation/podimpersonation.go

```diff
@@ -520,7 +493,7 @@ func (s *PodImpersonation) augmentPod(pod *v1.Pod, sa *v1.ServiceAccount, secret
 
 	pod = pod.DeepCopy()
 
-	pod.Spec.ServiceAccountName = ""
+	pod.Spec.ServiceAccountName = sa.Name
 	pod.Spec.AutomountServiceAccountToken = &f
```

Comment:
Should we consider using the `AutomountServiceAccountToken` feature and let k8s manage the token inject (volumes)? 
If I understand correctly, it was disabled because we created the Secret and would like to use one Secret for all pods.  
Now, We no longer need to maintain the Secret, so can we leave it to K8s? 
WDYT? Discussions are welcomed!

Reply:
I tested with that before, and it works, but I'm unsure if it handles rotating the tokens.

Reply:
Yes, K8s handles the rotation.
> In v1.22 and later, Kubernetes gets a short-lived, automatically rotating token using the TokenRequest API and mounts the token as a [projected volume](https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken).

Ref: https://kubernetes.io/docs/concepts/security/service-accounts/#assign-to-pod

 

I would like to hear more thoughts from the maintainer of the repo @rancher/rancher-team-1-neo-dev

Reply:
I'd agree with @jiaqiluo here, there's a lot of extra code add to project volumes that I would expect K8s to do for us. If there's a reason to manually set this up, happy to consider it, but for now I think we should default to the automounted value recommended by upstream.

Reply:
I agree as well, I wasn't aware that kubernetes would rotate the tokens automatically if using that flag, just that it does that when using a projected volume